lightning-custom-message/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+//! Utilities for supporting custom peer-to-peer messages in LDK.
+//!
+//! [BOLT 1] specifies a custom message type range for use with experimental or application-specific
+//! messages. While a [`CustomMessageHandler`] can be defined to support more than one message type,
+//! defining such a handler requires a significant amount of boilerplate and can be error prone.
+//!
+//! This crate provides the [`composite_custom_message_handler`] macro for easily composing
+//! pre-defined custom message handlers. The resulting handler can be further composed with other
```

Comment:
Could maybe be a bit clearer "composing" means composing together multiple custom message handlers together into one.